checkstyle-tester/projects-for-circle.properties

```diff
@@ -8,12 +8,12 @@ guava|git|https://github.com/google/guava|v28.2||
 #openjdk7|hg|http://hg.openjdk.java.net/jdk7/jdk7/jdk/|||
 #openjdk8|hg|http://hg.openjdk.java.net/jdk8/jdk8/jdk/|default|
 #All details at checkstyle/checkstyle#3033: ModalDialogActivationTest till JDK-8166015 ; 'jxc/8073519/**' not compilable by design ; ', jhsdb/**' - checkstyle do not support unicode identifiers
-#openjdk9|hg|http://hg.openjdk.java.net/jdk9/jdk9/jdk/|default|**/test/java/awt/Focus/ModalDialogActivationTest/ModalDialogActivationTest.java,**/test/javax/xml/bind/jxc/8073519/**,**/test/sun/tools/jhsdb/**
+#openjdk9|hg|http://hg.openjdk.java.net/jdk9/jdk9/jdk/|default|/test/java/awt/Focus/ModalDialogActivationTest/ModalDialogActivationTest.java,/test/javax/xml/bind/jxc/8073519/,/test/sun/tools/jhsdb/
 #infinispan|git|https://github.com/infinispan/infinispan|7.2.5.Final||
 #protonpack|git|https://github.com/poetix/protonpack|protonpack-1.7||
 #jOOL|git|https://github.com/jOOQ/jOOL|version-0.9.7||
 #lucene-solr|git|https://github.com/apache/lucene-solr|releases/lucene-solr/5.4.0||
-#tapestry-5|git|https://github.com/apache/tapestry-5|5.4.0|**/archetype-resources/**/*|
+#tapestry-5|git|https://github.com/apache/tapestry-5|5.4.0|/archetype-resources/|
```

Comment:
can we keep "**" ? it help to make it clear that it is some part in the middle.

Reply:
`**` is not valid regular expression.

Reply:
yes, I know , but it is still syntax of path matching

Reply:
Ok, how? Since it isn't a valid regular expression, it will cause an exception. If we forget it, it just proves how pointless it is.

````
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>

    <module name="TreeWalker">
<module name="Regexp">
  <property name="format" value="Example**"/>
</module>
    </module>
</module>
````

````
Caused by: java.util.regex.PatternSyntaxException: Dangling meta character '*' near index 8
Example**
        ^
    at java.base/java.util.regex.Pattern.error(Pattern.java:2028)
    at java.base/java.util.regex.Pattern.sequence(Pattern.java:2203)
    at java.base/java.util.regex.Pattern.expr(Pattern.java:2069)
    at java.base/java.util.regex.Pattern.compile(Pattern.java:1783)
    at java.base/java.util.regex.Pattern.<init>(Pattern.java:1429)
    at java.base/java.util.regex.Pattern.compile(Pattern.java:1095)
    at com.puppycrawl.tools.checkstyle.utils.CommonUtil.createPattern(CommonUtil.java:100)
    ... 15 more
````

Reply:
https://stackoverflow.com/a/22636142/1015848

Reply:
> Ant-style path patterns matching 

We are using CLI, not Ant or spring.

CLI doesn't support this type of matching, only regular expression.
See https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L785

Reply:
Yes, let's drop ant syntax.

But can you put in exclude value leading `.*` to show clearly that it is regular regexp.

Reply:
Done.